metrics_sdk/lib/opentelemetry/sdk/metrics/state/metric_store.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module State
+        class MetricStore
+          def initialize
+            @mutex = Mutex.new
+            @epoch_start_time = Time.now.to_i
+            @epoch_end_time = nil;
+            @metric_streams = []
+          end
+
+          def collect
+            # this probably needs to take the mutex, take a snapshot of state, (reset state?), release the mutex
+            @mutex.synchronize do
+              [
+                @end_time = Time.now.to_i,
+                @metric_streams,
+                @epoch_start_time = @end_time,
+              ]
+              @metric_streams
```

Comment:
I don't understand these 6 lines - did you mean to return the array, or `@metric_streams` ?

Reply:
Updated to hopefully be more understandable, will resolve the comment to clear up the discussion on the PR.